src/sentry/static/sentry/app/views/eventsV2/table/draggableColumns.tsx

```diff
@@ -0,0 +1,267 @@
+    left: void 0,
+    top: void 0,
+
+    draggingColumnIndex: void 0,
+    destinationColumnIndex: void 0,
+  };
+
+  previousUserSelect: UserSelectValues | null = null;
+  portal: HTMLElement | null = null;
+  dragGhostRef = React.createRef<HTMLDivElement>();
+
+  startColumnDrag = (
+    event: React.MouseEvent<SVGSVGElement, MouseEvent>,
+    initialColumnIndex: number
+  ) => {
+    const isDragging = this.state.isDragging;
+
+    if (isDragging || event.type !== 'mousedown') {
+      return;
+    }
+
+    // prevent the user from selecting things when dragging a column.
+
+    this.previousUserSelect = setBodyUserSelect({
+      userSelect: 'none',
+      MozUserSelect: 'none',
+      msUserSelect: 'none',
+    });
+
+    // attach event listeners so that the mouse cursor can drag anywhere
+
+    window.addEventListener('mousemove', this.onDragMove);
+    window.addEventListener('mouseup', this.onDragEnd);
+
+    this.setState({
+      isDragging: true,
+      draggingColumnIndex: initialColumnIndex,
+      left: event.pageX,
+      top: event.pageY,
+    });
+  };
+
+  onDragMove = (event: MouseEvent) => {
+    if (!this.state.isDragging || event.type !== 'mousemove') {
+      return;
+    }
+
+    if (this.dragGhostRef.current) {
+      // move the ghosted column title
+
+      const ghostDOM = this.dragGhostRef.current;
+      ghostDOM.style.left = `${event.pageX}px`;
+      ghostDOM.style.top = `${event.pageY}px`;
+    }
+
+    const gridHeadCellButtons = document.querySelectorAll(
+      `.${DRAGGABLE_COLUMN_CLASSNAME_IDENTIFIER}`
+    );
+
+    const destinationColumnIndex = Array.from(gridHeadCellButtons).findIndex(
+      headerElement => {
+        const rects = headerElement.getBoundingClientRect();
+
+        const left = event.pageX;
+
+        const thresholdStart = rects.left;
+        const thresholdEnd = rects.left + rects.width;
+
+        return left >= thresholdStart && left <= thresholdEnd;
+      }
+    );
+
+    if (destinationColumnIndex >= 0) {
+      if (this.state.destinationColumnIndex !== destinationColumnIndex) {
+        this.setDestinationColumnIndex(destinationColumnIndex);
+      }
+    }
+  };
+
+  setDestinationColumnIndex = debounce((destinationColumnIndex: number) => {
+    this.setState({
+      destinationColumnIndex,
+    });
+  }, 100);
+
+  onDragEnd = (event: MouseEvent) => {
+    if (!this.state.isDragging || event.type !== 'mouseup') {
+      return;
+    }
+
+    // remove listeners that were attached in startColumnDrag
+
+    this.cleanUpListeners();
+
+    // restore body styles
+
+    if (this.previousUserSelect) {
+      setBodyUserSelect(this.previousUserSelect);
+      this.previousUserSelect = null;
+    }
+
+    // indicate drag has ended
+
+    const destinationColumnIndex = this.state.destinationColumnIndex;
+    const draggingColumnIndex = this.state.draggingColumnIndex;
+
+    this.setState({
+      isDragging: false,
+      left: void 0,
```

Comment:
Would it be better to use `undefined`?

Reply:
They should mean the same thing. Either isn't better any than the other. 

I generally prefer `void 0` because it's shorter to type 😆 . Uglifyjs does this for the same reasons to trim as much characters as possible.